cds_ils/migrator/series/api.py

```diff
@@ -66,75 +72,109 @@ def create_multipart_volumes(pid, multipart_legacy_recid, migration_volumes):
         if volume_number not in volumes:
             volumes[volume_number] = {}
         volume = volumes[volume_number]
-        for key in obj:
-            if key != "volume":
-                if key in volume:
-                    raise KeyError(
-                        'Duplicate key "{}" for multipart {}'.format(
-                            key, multipart_legacy_recid
+        if "isbn" in obj:
+            if "isbns" not in volume:
+                volume["isbns"] = []
+            volume["isbns"].append({
+                "value": obj["isbn"],
+                "is_electronic": bool(obj["is_electronic"])
+            })
+            # TODO physical description
+        elif "barcode" in obj:
+            # this entry represents an item
+            if "items" not in volume:
+                volume["items"] = []
+            volume["items"].append({
+                "barcode": obj["barcode"]
+            })
+        else:
+            for key in obj:
+                if key != "volume":
+                    if key in volume:
+                        raise KeyError(
+                            'Duplicate key "{}" for multipart {}'.format(
+                                key, multipart_legacy_recid
+                            )
                         )
-                    )
-                volume[key] = obj[key]
+                    volume[key] = obj[key]
 
     volume_numbers = iter(sorted(volumes.keys()))
 
-    # Re-use the current record for the first volume
-    # TODO review this - there are more cases of multiparts
-    first_volume = next(volume_numbers)
-    first = Document.get_record_by_pid(pid)
-    if "title" in volumes[first_volume]:
-        first["title"] = volumes[first_volume]["title"]
-        first["volume"] = first_volume
-    first["_migration"]["multipart_legacy_recid"] = multipart_legacy_recid
+    blueprint = document_blueprint
```

Comment:
I wouldn't call it a blueprint because it means something specific in flask. draft maybe? document_temp_dict?